src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityChiefBarbarian.java

```diff
@@ -0,0 +1,118 @@
+import net.minecraft.entity.ai.*;
+import net.minecraft.entity.monster.EntityMob;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+/**
+ * Created by Asher on 5/6/17.
+ */
+public class EntityChiefBarbarian extends EntityMob
+{
+    private final       Colony           colony = ColonyManager.getClosestColony(world, this.getPosition());
+    public static final ResourceLocation LOOT   = new ResourceLocation(Constants.MOD_ID, "EntityChiefBarbarianDrops");
+
+    /**
+     * defines the default values for the Entity's attributes.
+     */
+    private static final double FOLLOW_RANGE          = 35.0D;
+    private static final double MOVEMENT_SPEED        = 0.2D;
+    private static final double ATTACK_DAMAGE         = 2.0D;
+    private static final double ARMOR                 = 2.0D;
+    private static final double BARBARIAN_BASE_HEALTH = 25;
+
+    /**
+     * Defines the default values for the various AI Task's priorities.
+     */
+    private static final int PRIORITY_ZERO   = 1;
+    private static final int PRIORITY_TWO   = 2;
+    private static final int PRIORITY_THREE = 3;
+    private static final int PRIORITY_FOUR = 4;
+    private static final int PRIORITY_FIVE = 5;
+    private static final int PRIORITY_SIX   = 6;
+    private static final int PRIORITY_EIGHT  = 8;
+
+    public EntityChiefBarbarian(final World worldIn)
+    {
+        super(worldIn);
+        this.getAlwaysRenderNameTag();
+    }
+
+    @Override
+    protected void applyEntityAttributes()
+    {
+        super.applyEntityAttributes();
+        this.getEntityAttribute(SharedMonsterAttributes.FOLLOW_RANGE).setBaseValue(FOLLOW_RANGE);
+        this.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED).setBaseValue(MOVEMENT_SPEED);
+        this.getEntityAttribute(SharedMonsterAttributes.ATTACK_DAMAGE).setBaseValue(ATTACK_DAMAGE);
+        this.getEntityAttribute(SharedMonsterAttributes.ARMOR).setBaseValue(ARMOR);
+        this.getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(this.getHealthBasedOnRaidLevel());
+    }
+
+    protected double getHealthBasedOnRaidLevel()
+    {
+        if (colony != null)
+        {
+            final int raidLevel = (int) (colony.getRaidLevel() * 1.5);
+            return BARBARIAN_BASE_HEALTH + raidLevel;
+        }
+        return BARBARIAN_BASE_HEALTH;
+    }
+
+    @Override
+    protected void initEntityAI()
+    {
+        this.tasks.addTask(PRIORITY_ZERO, new EntityAISwimming(this));
+        this.tasks.addTask(PRIORITY_FIVE, new EntityAIMoveTowardsRestriction(this, 1.0D));
+        this.tasks.addTask(PRIORITY_THREE, new EntityAIBarbarianAttackMelee(this));
+        this.tasks.addTask(PRIORITY_FOUR, new EntityAIWalkToRandomHuts(this, 2.0D));
+        this.tasks.addTask(PRIORITY_EIGHT, new EntityAIWatchClosest(this, EntityPlayer.class, 8.0F));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 8.0F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)